autobotAI_integrations/integrations/gitlab/__init__.py

```
@@ -157,3 +158,12 @@ def generate_cli_creds(self) -> CLICreds:
         return CLICreds(
             installer_check=installer_check, install_command=install_command, envs=envs
         )
+
+    def generate_mcp_creds(self) -> MCPCreds:
+        if str(self.integration.base_url).startswith("https://gitlab.com"):
+            return MCPCreds(
+                headers={
+                    "Authorization": f"Bearer {self.integration.token}",
+                },
+            )
+        raise Exception("Remote MCP is only supported for public gitlab.com")
```

Ruff (0.16.0)

[warning] Create your own exception

(TRY002)